### PR TITLE
Add tool to periodically report the coding completeness

### PR DESCRIPTION
**Pull request by lukechurch**
@marianamarasoiu PTAL

This is currently separated into tools to compute the fraction and a separate tool to push it, as they're in separate firebase databases for security reasons. There is some problem with switching the client authentication after initialisation, so I'm avoiding it using two processes for now. 

The intention is this tool would be executed approximately hourly, and the results pushed to a dashboard

### CODE REVIEW

File: data_tools/firebase_client_wrapper.py [hunk 1]
```diff
@@ -45,16 +46,18 @@ def get_message_ids(dataset_id):
         ids.append(message.id)
     return ids
 
-# This is a much faster way of doing an entire dataset read than repeated get_message calls
+# This is a much faster way of reading an entire dataset read than repeated get_message calls
```
Comment (marianamarasoiu) [inline data_tools/firebase_client_wrapper.py]: Well spotted opportunity for de-americanisation ;) 
Can you also please delete "read" later into the sentence (... reading and entire dataset **read** than ...)?

File: data_tools/firebase_client_wrapper.py [hunk 2]
```diff
@@ -45,16 +46,18 @@ def get_message_ids(dataset_id):
...
 
-
 def get_message(dataset_id, message_id):
     return client.document(u'datasets/{}/messages/{}'.format(dataset_id, message_id)).get().to_dict()
 
 def get_message_ref(dataset_id, message_id):
     return client.document(u'datasets/{}/messages/{}'.format(dataset_id, message_id))
+
+def push_coding_status(coding_status):
+    client.document(u'metrics/coda').set(coding_status)
```
Comment (marianamarasoiu) [inline data_tools/firebase_client_wrapper.py]: Can you also add a newline at the end of this file? Github shows a warning for it.

File: data_tools/push_coding_progress.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import firebase_client_wrapper as fcw
+
+import json
+import sys
+
+if (len(sys.argv) != 3):
+    print ("Usage python compute_coding_progress.py dashboard_crypto_token progress_file")
```
Comment (marianamarasoiu) [inline data_tools/push_coding_progress.py]: s/compute_coding_process.py/push_coding_process.py

File: data_tools/firebase_client_wrapper.py (hunk 1 above)
Comment (marianamarasoiu) [inline data_tools/firebase_client_wrapper.py]: Can you also please delete "read" later into the sentence (... reading and entire dataset read than ...)?

File: data_tools/firebase_client_wrapper.py (hunk 2 above)
Comment (lukechurch) [reply data_tools/firebase_client_wrapper.py]: done

File: data_tools/push_coding_progress.py (hunk 3 above)
Comment (lukechurch) [reply data_tools/push_coding_progress.py]: done

File: data_tools/firebase_client_wrapper.py (hunk 1 above)
Comment (lukechurch) [reply data_tools/firebase_client_wrapper.py]: done

**Comment by lukechurch** [comment]
Thanks @marianamarasoiu, review comments addressed

**Pull request merged**